nupic/regions/TPRegion.py

```diff
@@ -506,6 +517,18 @@ def _compute(self, inputs, outputs):
       size = activeLearnCells.shape[0] * activeLearnCells.shape[1]
       outputs['lrnActiveStateT'][:] = activeLearnCells.reshape(size)
 
+    if self.computePredictedActiveCellIndices:
+      # Reshape so we are dealing with 1D arrays
+      activeState = self._tfdr.getActiveState().reshape(-1).astype('float32')
+      predictedState = self._tfdr.getPredictedState().reshape(-1).astype('float32')
+      activeIndices = numpy.where(activeState != 0)[0]
+      predictedIndices= numpy.where(predictedState != 0)[0]
+      predictedActiveIndices = numpy.intersect1d(activeIndices, predictedIndices)
+      outputs["activeCells"].fill(0)
+      outputs["activeCells"][activeIndices] = 1
+      outputs["predictedActiveCells"].fill(0)
+      outputs["predictedActiveCells"][predictedActiveIndices] = 1
```

Comment:
I think "predictedActiveCells" is ambiguous at best, and totally obscuring
at worst - and is no where near as clear as successfullyPredictedCells. But
that's ultimately up to @ScottPurdy... I did however make this change in
HTM.java ;-)

On Mon, Jun 29, 2015 at 11:57 AM, Akhila <notifications@github.com> wrote:

> In nupic/regions/TPRegion.py
> <https://github.com/numenta/nupic/pull/2206#discussion_r33485701>:
>
> > @@ -506,6 +517,18 @@ def _compute(self, inputs, outputs):
> >        size = activeLearnCells.shape[0] * activeLearnCells.shape[1]
> >        outputs['lrnActiveStateT'][:] = activeLearnCells.reshape(size)
> >
> > +    if self.computePredictedActiveCellIndices:
> > +      # Reshape so we are dealing with 1D arrays
> > +      activeState = self._tfdr.getActiveState().reshape(-1).astype('float32')
> > +      predictedState = self._tfdr.getPredictedState().reshape(-1).astype('float32')
> > +      activeIndices = numpy.where(activeState != 0)[0]
> > +      predictedIndices= numpy.where(predictedState != 0)[0]
> > +      predictedActiveIndices = numpy.intersect1d(activeIndices, predictedIndices)
> > +      outputs["activeCells"].fill(0)
> > +      outputs["activeCells"][activeIndices] = 1
> > +      outputs["predictedActiveCells"].fill(0)
> > +      outputs["predictedActiveCells"][predictedActiveIndices] = 1
>
> do we need to change the name from predictedActiveCells to
> successfullyPredictedCells?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numenta/nupic/pull/2206/files#r33485701>.
>



-- 
*With kind regards,*

David Ray
Java Solutions Architect

*Cortical.io <http://cortical.io/>*
Sponsor of:  HTM.java <https://github.com/numenta/htm.java>

d.ray@cortical.io
http://cortical.io